backend/app/domain/services/base/context.py

```diff
@@ -190,6 +190,7 @@ def get_perdi_contexts(
         ]
         random.shuffle(all_models)
         selected_models = all_models[:number_of_samples]
+        print("Selected models were", selected_models)
```

Comment:
I understand the print for debugging, but we need to be careful about the cleaning that we need to do after we check everything is working fine